values.yaml

```diff
@@ -61,6 +61,14 @@ st2:
   # If set, st2.password always overrides any existing password.
   # If not set, the password is auto-generated on install and preserved across upgrades.
   #password: Ch@ngeMe
+
+  ## Configure the st2 auth secret generated/used
+  auth_secret:
+    # Disable the creation of the secret to allow customers to manage the secret separately
+    create: true
+    # Set custom name of the secret for the ST2 Auth configuration
+    name: ""
```

Comment:
2 settings for this is a little too complex I think. Is there a reason to have a bool separate from the name?

If the chart owns the secret, it should follow the chart's naming convention. If the user provides the name, then the chart should not touch that secret, just use it.

Maybe call the value `auth_secret_name`. If it is an empty string, then generate the secret name and generate the secret. If it is not empty, then just use the name as is and skip generating the secret.

Reply:
I adjusted to having a single value, like you suggested. I went with the name `authExistingSecret`. 
This is based on how Bitnami Helm charts appear to do it, [example](https://github.com/bitnami/charts/tree/main/bitnami/postgresql-ha#global-parameters).

Please let me know what you think